natss/pkg/dispatcher/dispatcher.go

```diff
@@ -176,7 +162,7 @@ func (s *SubscriptionsSupervisor) Start(ctx context.Context) error {
 	go s.Connect(ctx.Done())
 	// Trigger Connect to establish connection with NATS
 	s.signalReconnect()
-	return s.ceClient.StartReceiver(ctx, s.receiver.ServeHTTP)
+	return s.receiver.Start(ctx)
 }
 
 func (s *SubscriptionsSupervisor) connectWithRetry(stopCh <-chan struct{}) {
```

Comment:
you should wire up context in this method and pass it to subscribe so the subscriber can shutdown on sigterm signal.